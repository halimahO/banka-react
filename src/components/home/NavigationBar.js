import React from 'react';

const NavigationBar = () => {
  return <div>I am the nav bar</div>;
  // <div className="container">
  //   <div id="branding">
  //     <a href="index.html">
  //       <img id="branding" src="./img/logo-blue.png" alt="logo-white" />
  //     </a>
  //   </div>
  //   <nav>
  //     <div className="nav-full">
  //       <div className="nav-style navbar">
  //         <ul>
  //           <li className="current">
  //             <a href="index.html">Home</a>
  //           </li>
  //           <li>
  //             <a href="sign-in.html">Sign in</a>
  //           </li>
  //           <li>
  //             <a href="register.html">Register</a>
  //           </li>
  //         </ul>
  //       </div>
  //     </div>
  //     <div className="nav-not-full">
  //       <div className="hidden-navbar">
  //         {/* <img
  //           id="menu"
  //           className="nav-style"
  //           src="./img/menu.png"
  //           alt=""
  //           onClick="toggleSidebar()"
  //         /> */}
  //         <div id="sidebar">
  //           <ul>
  //             <li className="current">
  //               <a href="index.html">Home</a>
  //             </li>
  //             <li>
  //               <a href="sign-in.html">Sign in</a>
  //             </li>
  //             <li>
  //               <a href="register.html">Register</a>
  //             </li>
  //           </ul>
  //         </div>
  //       </div>
  //     </div>
  //   </nav>
  // </div>
};
export default NavigationBar;
