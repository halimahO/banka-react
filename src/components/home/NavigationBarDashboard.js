/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import navBarLogo from '../../img/logo-blue.png';
import menu from '../../img/menu.png';

class NavigationBarDashboard extends React.Component {
  constructor() {
    super();
    this.sidebar = React.createRef();
  }

  toggleSidebar = () => {
    const sidebar = this.sidebar.current;
    if (sidebar.style.display === 'none') {
      sidebar.style.display = 'block';
    } else {
      sidebar.style.display = 'none';
    }
  };

  render() {
    const lists = document.querySelectorAll('.menu-items li');
    const removeClass = () => {
      const activeMenu = document.querySelector(
        '.dashboard-area .items.active'
      );
      activeMenu.classList.remove('active');
    };

    const showCurrentMenu = menuId => {
      document.getElementById(menuId).classList.add('active');
    };

    Array.from(lists).forEach(listItem => {
      listItem.addEventListener('click', event => {
        const tartgetSectionId = event.target.dataset.menu;
        removeClass();
        showCurrentMenu(tartgetSectionId);
        document.getElementById(
          'menuTopic'
        ).innerHTML = tartgetSectionId.replace(/-/g, ' ');
      });
    });
    return (
      <div className="shadow">
        <div className="container">
          <div id="branding">
            <Link to="/">
              <img className="branding" src={navBarLogo} alt="logo-blue" />
            </Link>
          </div>
          <nav>
            <div className="nav-full">
              <div className="nav-style navbar">
                <ul>
                  <li>
                    <Link to="/">Home</Link>
                  </li>
                  <li>
                    <Link to="/dashboard">Profile</Link>
                  </li>
                  <li>
                    <Link to="/create-account">Open an Account</Link>
                  </li>
                  <li>
                    <Link to="/">Log out</Link>
                  </li>
                </ul>
              </div>
            </div>
            <div className="nav-not-full">
              <div className="hidden-navbar">
                <img
                  id="menu"
                  className="nav-style menuToggle"
                  src={menu}
                  alt=""
                  onClick={this.toggleSidebar}
                />
                <div ref={this.sidebar} id="sidebar">
                  <ul>
                    <li>
                      <Link to="/">Home</Link>
                    </li>
                    <li>
                      <Link to="/dashboard">Profile</Link>
                    </li>
                    <li>
                      <Link to="/create-account">Open an Account</Link>
                    </li>
                    <div className="menu-items dashboard-menu">
                      <li
                        data-menu="account-history"
                        onClick={this.toggleSidebar}
                      >
                        Account History
                      </li>
                      {this.props.auth.user.type !== 'client' ? (
                        <React.Fragment>
                          <li
                            data-menu="debit-account"
                            onClick={this.toggleSidebar}
                          >
                            Debit Account
                          </li>
                          <li
                            data-menu="credit-account"
                            onClick={this.toggleSidebar}
                          >
                            Credit Account
                          </li>
                        </React.Fragment>
                      ) : null}
                    </div>
                    <li>
                      <Link to="/">Log out</Link>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </nav>
        </div>
      </div>
    );
  }
}

NavigationBarDashboard.propTypes = {
  history: PropTypes.object,
  isAuthenticated: PropTypes.bool,
  auth: PropTypes.object
};

export const mapStateToProps = state => {
  return {
    auth: state.auth
  };
};

export default connect(mapStateToProps)(NavigationBarDashboard);
