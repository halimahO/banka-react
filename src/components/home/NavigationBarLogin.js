/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import { Link } from 'react-router-dom';
import navBarLogo from '../../img/logo-blue.png';
import menu from '../../img/menu.png';

const NavigationBar = () => {
  const toggleSidebar = () => {
    const sidebar = document.getElementById('sidebar');
    if (sidebar.style.display === 'none') {
      sidebar.style.display = 'block';
    } else {
      sidebar.style.display = 'none';
    }
  };
  return (
    <div className="shadow">
      <div className="container">
        <div id="branding">
          <Link to="/">
            <img className="branding" src={navBarLogo} alt="logo-blue" />
          </Link>
        </div>
        <nav>
          <div className="nav-full">
            <div className="nav-style navbar">
              <ul>
                <li className="current">
                  <Link to="/">Home</Link>
                </li>
                <li>
                  <Link to="/login">Sign in</Link>
                </li>
                <li>
                  <Link to="/signup">Register</Link>
                </li>
              </ul>
            </div>
          </div>
          <div className="nav-not-full">
            <div className="hidden-navbar">
              <img
                id="menu"
                className="nav-style"
                src={menu}
                alt=""
                onClick={toggleSidebar}
              />
              <div id="sidebar">
                <ul>
                  <li className="current">
                    <Link to="/">Home</Link>
                  </li>
                  <li>
                    <Link to="/login">Sign in</Link>
                  </li>
                  <li>
                    <Link to="/signup">Register</Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </nav>
      </div>
    </div>
  );
};
export default NavigationBar;
