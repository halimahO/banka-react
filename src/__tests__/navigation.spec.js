import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter as Router, BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import configMockStore from 'redux-mock-store';
import NavigationBar from '../components/home/NavigationBar';
import NavigationBarDashboard from '../components/home/NavigationBarDashboard';
import NavigationBarLogin from '../components/home/NavigationBarLogin';

const initialState = {
  isAuthenticated: false,
  user: null
};

const mockStore = configMockStore([thunk]);
const store = mockStore(initialState);

const props = {
  logout: jest.fn(),
  auth: {
    isAuthenticated: false,
    user: {
      firstname: 'name',
      lastname: 'name',
      type: 'client',
      token: 'token'
    }
  },
  history: {
    push: jest.fn()
  }
};

const wrapper = shallow(
  <Provider store={store}>
    <Router>
      <NavigationBar {...props} />
    </Router>
  </Provider>
);

describe('<NavigationBar/> rendering', () => {
  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should simulate logout', () => {
    props.auth.isAuthenticated = true;
    const event = {
      preventDefault: jest.fn()
    };

    const component = mount(
      <BrowserRouter>
        <NavigationBar.WrappedComponent {...props} />
      </BrowserRouter>
    );
    component.find('#logout').simulate('click', event);
    expect(props.logout).toHaveBeenCalled();
    expect(props.history.push).toHaveBeenCalled();
    expect(event.preventDefault).toHaveBeenCalled();
    props.auth.isAuthenticated = false;
  });

  it('should simulate sidebar toggle', () => {
    const component = mount(
      <BrowserRouter>
        <NavigationBar.WrappedComponent {...props} />
      </BrowserRouter>
    );
    const instance = component.find('NavigationBar').instance();
    const spy = jest.spyOn(instance, 'toggleSidebar');
    instance.forceUpdate();
    component.find('#menu').simulate('click');

    expect(spy).toHaveBeenCalled();
  });
});

const wrapperDashboard = shallow(
  <Provider store={store}>
    <Router>
      <NavigationBarDashboard {...props} />
    </Router>
  </Provider>
);

describe('<NavigationBarDashboard/> rendering', () => {
  it('should render without crashing', () => {
    expect(wrapperDashboard).toMatchSnapshot();
  });

  it('should simulate sidebar toggle', () => {
    const component = mount(
      <BrowserRouter>
        <NavigationBarDashboard.WrappedComponent {...props} />
      </BrowserRouter>
    );
    const instance = component.find('NavigationBarDashboard').instance();
    const spy = jest.spyOn(instance, 'toggleSidebar');
    instance.forceUpdate();
    component.find('.menuToggle').simulate('click');

    expect(spy).toHaveBeenCalled();
  });
});

const wrapperLogin = shallow(
  <Provider store={store}>
    <Router>
      <NavigationBarLogin {...props} />
    </Router>
  </Provider>
);

describe('<NavigationBarLogin/> rendering', () => {
  it('should render without crashing', () => {
    expect(wrapperLogin).toMatchSnapshot();
  });

  it('should simulate sidebar toggle', () => {
    const component = mount(
      <BrowserRouter>
        <NavigationBarLogin.WrappedComponent {...props} />
      </BrowserRouter>
    );
    const instance = component.find('NavigationBarLogin').instance();
    const spy = jest.spyOn(instance, 'toggleSidebar');
    instance.forceUpdate();
    component.find('.menuSidebar').simulate('click');

    expect(spy).toHaveBeenCalled();
  });
});
