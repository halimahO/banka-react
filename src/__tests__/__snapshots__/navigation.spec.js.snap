// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar/> rendering should render without crashing 1`] = `ShallowWrapper {}`;

exports[`<NavigationBarDashboard/> rendering should render without crashing 1`] = `ShallowWrapper {}`;

exports[`<NavigationBarLogin/> rendering should render without crashing 1`] = `ShallowWrapper {}`;
